synapse/lib/httpapi.py

```diff
@@ -149,45 +151,139 @@ async def reqAuthUser(self):
+        if iden is None:
             self.sendAuthRequired()
             return False
 
-        if not user.isAdmin():
-            self.sendRestErr('AuthDeny', f'User {user.iden} ({user.name}) is not an admin.')
+        authcell = self.getAuthCell()
+        udef = await authcell.getUserDef(iden, packroles=False)
+        if not udef.get('admin'):
+            self.sendRestErr('AuthDeny', f'User {self.web_useriden} ({self.web_username}) is not an admin.')
             return False
 
         return True
 
     async def sess(self, gen=True):
+        '''
+        Get the heavy Session object for the request.
+
+        Args:
+            gen (bool): If set to True, generate a new session if there is no sess cookie.
+
+        Notes:
+            This stores the identifier in the ``sess`` cookie for with a 14 day expiration, stored
+            in the Cell.
+            Valid requests with that ``sess`` cookie will resolve to the same Session object.
+
+        Returns:
+            Sess: A heavy session object. If the sess cookie is invalid or gen is false, this returns None.
+        '''
 
         if self._web_sess is None:
 
             iden = self.get_secure_cookie('sess', max_age_days=14)
 
-            if iden is None and not gen:
-                return None
-
             if iden is None:
-                iden = s_common.guid().encode()
-                opts = {'expires_days': 14, 'secure': True, 'httponly': True}
-                self.set_secure_cookie('sess', iden, **opts)
+                if gen:
+                    iden = s_common.guid().encode()
+                    opts = {'expires_days': 14, 'secure': True, 'httponly': True}
+                    self.set_secure_cookie('sess', iden, **opts)
+                else:
+                    return None
 
             self._web_sess = await self.cell.genHttpSess(iden)
 
         return self._web_sess
 
-    async def user(self):
+    async def user(self):  # pragma: no cover
+        '''
+        Get the heavy User object from the Cell for the authenticated user.
+
+        Deprecated, use the useriden() function instead.
+        '''
+        s_common.deprecated('HandlerBase.user() - use useriden()', eolv='v2.140.0')  # set curv?
+        iden = await self.useriden()
+        if iden is None:
+            return None
+        user = self.cell.auth.user(iden)
+        self._web_user = user
+        return user
+
+    async def useriden(self):
+        '''
+        Get the user iden of the current session user.
 
-        if self._web_user is not None:
-            return self._web_user
+        Note:
+            This function will pull the iden from the current session, or
+            attempt to resolve the useriden with basic authentication.
+
+        Returns:
+            str: The iden of the current session user.
+        '''
+        authcell = self.getAuthCell()
+
+        if self.web_useriden is not None:
+            return self.web_useriden
```

Comment:
While we're in here, i think we should consider switching to the session object as the sole source of truth about the user iden/name. It feels like we're doing a lot of stashing those values on handlers and doing a bit too much gymnastics?

Reply:
We could make that change ( requiring all handlers to use a Sess object ); but that would also mean that a request  which uses basic auth would need a Sess object as well. That would either need to be ephemeral or we are adding a nexus transaction per-request just to do the authentication.